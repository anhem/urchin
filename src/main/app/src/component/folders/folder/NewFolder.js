import React, {Component} from 'react'
import FieldError from "../../FieldError"
import history from '../../../history'

class NewFolder extends Component {

    update = (e) => {
        let data = {
            field: e.target.name,
            value: e.target.value
        };

        this.props.callbacks.setFolder(data);
    };

    create = () => {
        this.props.callbacks.createFolder(this.props.folder);
    };

    cancel = () => {
        history.push('/folders');
    };

    render() {
        let folder = this.props.folder.folder || '';

        return (
            <div id="new-folder">
                <h2>Folder</h2>
                <input
                    name="folder"
                    type="text"
                    value={folder}
                    onChange={this.update}
                />
                <FieldError
                    fieldErrors={this.props.fieldErrors}
                    field="folder"
                />
                <button id="new-folder__create-btn"  onClick={this.create}>Create Folder</button>
                <button id="new-folder__cancel-btn" onClick={this.cancel}>Cancel</button>
            </div>
        )
    }
}

export default NewFolder