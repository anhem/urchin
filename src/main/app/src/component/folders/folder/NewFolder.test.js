import React from 'react';
import {shallow} from 'enzyme'
import toJson from 'enzyme-to-json';
import NewFolder from "./NewFolder";

describe('NewFolder', () => {

    let props = {
        folder: {},
    };

    it('match snapshot', () => {
        expect(toJson(shallow(<NewFolder {...props}/>))).toMatchSnapshot();
    });
});
