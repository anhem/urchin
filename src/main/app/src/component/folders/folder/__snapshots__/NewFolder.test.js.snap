// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewFolder match snapshot 1`] = `
<div
  id="new-folder"
>
  <h2>
    Folder
  </h2>
  <input
    name="folder"
    onChange={[Function]}
    type="text"
    value=""
  />
  <FieldError
    field="folder"
  />
  <button
    id="new-folder__create-btn"
    onClick={[Function]}
  >
    Create Folder
  </button>
  <button
    id="new-folder__cancel-btn"
    onClick={[Function]}
  >
    Cancel
  </button>
</div>
`;
